tests/tensorflow/test_tensorflow2_autolog.py

```diff
@@ -260,6 +261,111 @@ def test_tf_keras_autolog_logs_expected_data(tf_keras_random_data_run):
     assert "model_summary.txt" in artifacts
 
 
+def __example_tf_dataset(batch_size):
+    a = tf.data.Dataset.range(1)
+    b = tf.data.Dataset.range(1)
+    ds = tf.data.Dataset.zip((a, b))
+    return ds.batch(batch_size)
+
+
+class __ExampleSequence(tf.keras.utils.Sequence):
+    def __init__(self, batch_size):
+        self.batch_size = batch_size
+
+    def __len__(self):
+        return 10
+
+    def __getitem__(self, idx):
+        return np.array([idx] * self.batch_size), np.array([-idx] * self.batch_size)
+
+
+def __generator(data, target, batch_size):
+    data_batches = np.split(data, data.shape[0] // batch_size)
+    target_batches = np.split(target, target.shape[0] // batch_size)
+    for data_batch, target_batch in zip(data_batches, target_batches):
+        yield data_batch, target_batch
+
+
+class __GeneratorClass:
+    def __init__(self, data, target, batch_size):
+        self.data = data
+        self.target = target
+        self.batch_size = batch_size
+        self.ptr = 0
+
+    def __next__(self):
+        if self.ptr >= len(self.data):
+            raise StopIteration
+        idx = self.ptr % len(self.data)
+        self.ptr += 1
+        return self.data[idx : idx + self.batch_size], self.target[idx : idx + self.batch_size]
+
+    def __iter__(self):
+        return self
+
+
+@pytest.mark.large
+@pytest.mark.parametrize(
+    "generate_data",
+    [
+        __example_tf_dataset,
+        __ExampleSequence,
+        functools.partial(__generator, np.array([[1]] * 10), np.array([[1]] * 10)),
+        functools.partial(__GeneratorClass, np.array([[1]] * 10), np.array([[1]] * 10)),
+    ],
+)
+@pytest.mark.parametrize("batch_size", [5, 10])
+def test_tf_keras_autolog_implicit_batch_size_works(generate_data, batch_size):
+    mlflow.autolog()
+    model = tf.keras.Sequential()
+    model.add(tf.keras.layers.Dense(1, input_shape=(1,)))
+    model.compile(loss="mse")
+
+    # 'x' passed as arg
+    model.fit(generate_data(batch_size), verbose=0)
+    assert mlflow.last_active_run().data.params["batch_size"] == str(batch_size)
+
+    # 'x' passed as kwarg
+    model.fit(x=generate_data(batch_size), verbose=0)
+    assert mlflow.last_active_run().data.params["batch_size"] == str(batch_size)
+
+
+@pytest.mark.large
+@pytest.mark.skipif(
+    Version(tf.__version__) < Version("2.1.4"),
+    reason="Does not support passing of generator classes as `x` in `fit`",
+)
+@pytest.mark.parametrize("generator", [__generator, __GeneratorClass])
```

Comment:
Are there any other types of generators used with tf.keras that we should test?

Reply:
There exist tf.keras built-in generators like: [ImageDataGenerator](https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image/ImageDataGenerator?version=nightly) and [TimeseriesGenerator](https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/sequence/TimeseriesGenerator?version=nightly). But they all inherits from [Sequence](https://www.tensorflow.org/api_docs/python/tf/keras/utils/Sequence) or [Iterator](https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image/Iterator?version=nightly) (which also itself inherits from [Sequence](https://www.tensorflow.org/api_docs/python/tf/keras/utils/Sequence)) which are already being tested. Additionally, they will soon be deprecated in future versions:

> Prefer loading data with either [tf.keras.utils.text_dataset_from_directory](https://www.tensorflow.org/api_docs/python/tf/keras/utils/text_dataset_from_directory) or [tf.keras.utils.image_dataset_from_directory](https://www.tensorflow.org/api_docs/python/tf/keras/utils/image_dataset_from_directory), and then transforming the output [tf.data.Dataset](https://www.tensorflow.org/api_docs/python/tf/data/Dataset) with preprocessing layers.

[Source](https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing?version=nightly)

Reply:
Thanks!